Add documentation for METRICS_INFO and TS_INFO (#145964)
elastic/elasticsearch: docs/reference/query-languages/esql/_snippets/commands/layout/metrics-info.md

```yaml {applies_to}

> serverless: ga 9.4.0

Reviewer: ```suggestion
serverless: ga
```

stack: ga 9.4.0
```

> The `METRICS_INFO` [processing command](/reference/query-languages/esql/commands/processing-commands.md) returns one

Reviewer 2: I think we need to call out that this is only relevant for the `TS` source, and returns errors if applied under `FROM` or even after `STATS` under `TS`. EDIT: I see, this is included under limitations. Sounds good.

Reviewer 2: I'd also probably start with a more generic description of the purpose of this command. I think it can be used to retrieve information about available metrics in a time-series data streams, along with applicable dimensions and other metadata.

row per distinct metric in a [time series data stream](docs-content://manage-data/data-store/data-streams/time-series-data-stream-tsds.md),
with metadata about each metric.

## Syntax

```esql
METRICS_INFO
```

## Parameters

> `METRICS_INFO` takes no parameters.

Reviewer: ```suggestion
:::{note}
`METRICS_INFO` takes no parameters.
:::
```

`metric_name`
:   The name of the metric field (single-valued).

`data_stream`
:   The data stream(s) that contain this metric (multi-valued when the metric spans multiple data streams).

`unit`
:   The unit declared in the field mapping, such as `bytes` or `packets` (multi-valued when backing indices
    differ; may be `null` if no unit is declared).

`metric_type`
:   The metric type, for example `counter` or `gauge` (multi-valued when definitions differ across backing indices).

`field_type`
:   The Elasticsearch field type, for example `long`, `double`, or `integer` (multi-valued when definitions differ).

`dimension_fields`
:   The dimension field names associated with this metric (multi-valued).

### Restrictions

- `METRICS_INFO` can only be used after a [`TS`](/reference/query-languages/esql/commands/ts.md) source command.
  Using it after `FROM` or other source commands produces an error.
- `METRICS_INFO` must appear before pipeline-breaking commands such as
  [`STATS`](/reference/query-languages/esql/commands/stats-by.md),
  [`SORT`](/reference/query-languages/esql/commands/sort.md), or
  [`LIMIT`](/reference/query-languages/esql/commands/limit.md).

> - The output replaces the original table — downstream commands operate on the metadata rows, not the

Reviewer: ```suggestion
- The output replaces the original table: downstream commands operate on the metadata rows, not the
```

  raw time series documents.

> ## Examples

Reviewer: we should try to have a short intro to each example just for consistency :)

information about the metrics available in
[time series data streams](docs-content://manage-data/data-store/data-streams/time-series-data-stream-tsds.md),
along with their applicable dimensions and other metadata.

Use `METRICS_INFO` to discover which metrics exist, what types and units they have, and which
dimensions apply to them without having to inspect index mappings or rely on the field
capabilities API. Any [`WHERE`](/reference/query-languages/esql/commands/where.md) filters that precede
`METRICS_INFO` narrow the set of time series considered, so only metrics with matching data are
returned.

## Syntax

```esql
METRICS_INFO
```

## Parameters

:::{note}
`METRICS_INFO` takes no parameters.
:::

## Description

`METRICS_INFO` produces one row per distinct metric signature — that is, per unique combination
of metric name and its properties across backing indices. When the same metric is defined with
different properties (for example, different units) in different data streams, separate rows are
returned for each variant.

The output contains the following columns, all of type `keyword`:

`metric_name`
:   The name of the metric field (single-valued).

`data_stream`

> :   The data stream(s) that contain this metric (multi-valued when the metric spans multiple data streams).

Reviewer 3: ```suggestion
:   The data stream(s) that contain this metric (multi-valued when the metric is included in multiple data streams which align on the unit, metric type, and field type).
```

`unit`
:   The unit declared in the field mapping, such as `bytes` or `packets` (multi-valued when backing indices

>     differ; may be `null` if no unit is declared).

Reviewer 3: ```suggestion
:   The unit declared in the field mapping, such as `bytes` or `packets` (multi-valued when definitions differ across backing indices;
   may be `null` if no unit is declared).
```

also add a link to https://www.elastic.co/docs/reference/elasticsearch/mapping-reference/mapping-field-meta

`metric_type`
:   The metric type, for example `counter` or `gauge` (multi-valued when definitions differ across backing indices).

`field_type`

> :   The Elasticsearch field type, for example `long`, `double`, or `integer` (multi-valued when definitions differ).

Reviewer 3: ```suggestion
:   The Elasticsearch field type, for example `long`, `double`, or `integer` (multi-valued when definitions differ across backing indices).
```

    such as `bytes` or `packets` (multi-valued when definitions differ across backing indices;
    may be `null` if no unit is declared).

`metric_type`
:   The metric type, for example `counter` or `gauge` (multi-valued when definitions differ across backing indices).

`field_type`
:   The Elasticsearch field type, for example `long`, `double`, or `integer` (multi-valued when definitions differ
    across backing indices).

`dimension_fields`
:   The dimension field names associated with this metric (multi-valued). The union of dimension
    keys across all time series for that metric.

### Restrictions

- `METRICS_INFO` can only be used after a [`TS`](/reference/query-languages/esql/commands/ts.md) source command.
  Using it after `FROM` or other source commands produces an error.
- `METRICS_INFO` must appear before pipeline-breaking commands such as
  [`STATS`](/reference/query-languages/esql/commands/stats-by.md),
  [`SORT`](/reference/query-languages/esql/commands/sort.md), or
  [`LIMIT`](/reference/query-languages/esql/commands/limit.md).
- The output replaces the original table: downstream commands operate on the metadata rows, not the
  raw time series documents.

> ## Examples

Reviewer 3: Ideally, the examples should be powered by CSV tests and show the results as well, similar to what we have for other commands. But this can be done in a follow-up.

Author: That's a good point, I replaced the examples with the queries from csv tests
